src/yang/ietf-bgp.yang

```diff
@@ -238,6 +243,66 @@ module ietf-bgp {
     uses rt-pol:apply-policy-group;
   }
 
+  grouping bgp-errors-common {
+    description
+      "BGP NOTIFICATION state that is common in the sent and received
+       direction.";
+      
+    leaf last-notification {
+      type yang:date-and-time;
+      description
+        "The timestamp indicates the time that a BGP
+         NOTIFICATION message was last handled.";
+      reference
+        "RFC 4271, Section 4.5.";
+    }
+    leaf last-error {
+      type identityref {
+        base bn:bgp-notification;
+      }
+      description
+        "The last NOTIFICATION error for the peer on this
+         connection.  If no identity is registered for the
+         Error code / Error subcode, the last-error-code and
+         last-error-subcode will have the information received
+         in the NOTIFICATION PDU.";
+      reference
+        "RFC 4271, Section 4.5.";
+    }
+    leaf last-error-code {
+      type uint8;
+      description
+        "The last NOTIFICATION Error code for the peer on
+         this connection.";
+      reference
+        "RFC 4271, Section 4.5.";
+    }
+    leaf last-error-subcode {
+      type uint8;
+      description
+        "The last NOTIFICATION Error subcode for the peer on
+         this connection.";
+      reference
+        "RFC 4271, Section 4.5.";
+    }
+  }
+  grouping bgp-errors-common-data {
+    description
+      "BGP NOTIFICATION data state that is common in the sent and
+       received direction.";
+
+    leaf last-error-data {
+      type binary {
```

Comment:
last-error-data isn't the code/sub-code, it's the "value" portion of the notification TLV outside of those points.  The data will be inherently opaque without decoding.

I'd rather stick to binary types for this class of stuff than use something like hex-string because hex-string would require a converter just to get it back to the binary type.  While I agree that there may be some circumstances where being able to glance at raw xml and see the data that's hiding in the base64 is useful, I think that's a low-value proposition.